### ISSUE TITLE
Training model with over 100 intents

### ISSUE THREAD

**Issue by kelvinacc** [RasaHQ/rasa_nlu#924]
Rasa NLU version - 0.11.3

Used backend / pipeline - spacy

Operating system - Windows

I have a few questions regarding how to train the a model with multiple intents.

1. What is the best way to train abbreviations or single words? e.g. when someone only says NLP (Natural Language Processing) as an expression.

2. As we have over a 100 intents, what is the best way to train the model, any techniques to use to ensure accuracy.

3. How does the NLP calculate the confidence of the intent as sometimes it picks a wrong intent which does not contain a word of the expression in the training data.

Thanks

**Comment by tmbo** [comment]
You should train abbreviations just like normal text. One thing to look out for: they are usually not part of the word vector space and therefore their influence on the intent classification is limited - we are currently working on a model that can get around this limitation. Otherwise, it is good to add the `intent_featurizer_ngrams` to the processing pipeline.

Best is to train the model and take a look at the evaluation to see which intents get confused. This will provide you insights: 1. where to collect more training data 2. if it makes sense to restructure your intents

The meaning of the confidence scores depends on the underlying ML model used. For the sklearn classifier it is the distance from the hyperplane separating the classes. We don't use similarity based models (which compute the similarity of a test sentence with the training data), depending on how much training data you have, an SVM can be wrong while at the same time producing a high confidence.
